ospfd/ospf_vty.c

```diff
@@ -7155,9 +7156,12 @@ DEFUN (show_ip_ospf_database_max,
 		} else {
 			ospf = ospf_lookup_by_inst_name(inst, vrf_name);
 			if (ospf == NULL || !ospf->oi_running) {
-				vty_out(vty,
-					"%% OSPF is not enabled in vrf %s\n",
-					vrf_name);
+				if (!uj)
+					vty_out(vty,
+						"%% OSPF is not enabled in vrf %s\n",
+						vrf_name);
+				else
```

Comment:
While I agree this fixes the memory leak I don't think it is the right fix for this case: ideally we should return a JSON string with some kind of error unless I'm missing something here. Example:
```js
{
  "error": "OSPF is not enabled in VRF foobar"
}
```

Do we want to fix the behavior or just plug the memory leak? Maybe create an accord to standardize the JSON error handling and move this issue somewhere else?

Reply:
I think we need agreement on how to handle this problem.  But it's not what I was going for here.  I can just remove this bit of code for the moment.

Incidentally there are several different ways this is already handled.

Reply:
removed the change here and just freed the memory that was being leaked